### DISCUSSION TITLE
How many wallets allow communication over Ports instead of window attributes?

### DISCUSSION THREAD

**Discussion by javaspeak** [WalletConnect/walletconnect-monorepo#2475] [Developer Support]
I would like to know how many wallets are copying Meta Mask functionality and have a provider which can connect over ports instead of communicating by injecting stuff like window.ethereum into the window context.

The problem of browser extensions is that they cannot see any attributes that were injected in the host window context (when inside the extension window context).

Some people working with browser extensions inject the wallet connect functionality as a script into the host website.  The problem is the security policy of the host website can disallow this which means we can only connect to the wallet in that situation if the wallet exposes its API over a port provider.

In short does anyone know which wallets copy Meta Mask and have a provider which allows for connection over port instead of the windows context?

**Comment by javaspeak** [comment]
Can anyone answer the above question? - thanks

**Comment by xzilja** [comment]
On our end we don't have this info I'm afraid.
